apps/chat/src/store/conversations/conversations.epics.ts

```diff
@@ -2412,6 +2389,74 @@ const updateConversationEpic: AppEpic = (action$, state$) =>
             EMPTY,
           ),
         ),
+        of(
+          ConversationsActions.updateConversationSuccess({
+            id,
+            conversation: {
+              ...values,
+              id: newConversation.id,
+            },
+          }),
+        ),
+      );
+    }),
+  );
+
+const updateLocalConversationEpic: AppEpic = (action$, state$) =>
+  action$.pipe(
+    filter(ConversationsActions.updateConversation.match),
+    filter(({ payload }) => isEntityIdLocal(payload)),
+    mergeMap(({ payload }) => {
+      const { id, values } = payload;
+
+      const conversation = ConversationsSelectors.selectConversation(
+        state$.value,
+        id,
+      ) as Conversation;
+
+      if (!conversation) {
+        return of(
+          UIActions.showErrorToast(
+            translate(
+              'It looks like this conversation has been deleted. Please reload the page',
+            ),
+          ),
+        );
+      }
+
+      const saveInStorage =
+        (values.isMessageStreaming === false &&
+          !!(values.messages?.length || conversation.messages?.length)) ||
+        (!!values.folderId &&
+          values.folderId !== getConversationRootId(LOCAL_BUCKET));
```

Comment:
```suggestion
const hasMessages =
  (values.messages && values.messages.length) ||
  (conversation.messages && conversation.messages.length);

const isInDifferentFolder =
  !!values.folderId &&
  values.folderId !== getConversationRootId(LOCAL_BUCKET);

const shouldSaveInStorage =
  values.isMessageStreaming === false && (hasMessages || isInDifferentFolder);
```

maybe lets do something like this to improve readability?